backend/inspirehep/orcid/domain_models.py

```diff
@@ -204,16 +206,30 @@ def _post_or_put_work(self, putcode=None):
         response.raise_for_result()
         return response["putcode"]
 
+    def _delete_works_with_duplicated_putcodes(self, cached_putcodes_recids):
+        unique_putcodes_recids = {}
+        for fetched_putcode, fetched_recid in cached_putcodes_recids:
+            if fetched_recid in unique_putcodes_recids.values():
```

Comment:
accessing a dictionary by keys is fast (O(1)), accessing a dictionary by values is slow (O(len(dict))). So it's better to invert the dict here so you can do
```suggestion
            if fetched_recid in unique_recids_putcodes:
```